### PR TITLE
Fixes #12969 - font-awesome-sass package for Patternfly

### PR DESCRIPTION
**Pull request by dLobatog**
https://rubygems.org/gems/patternfly-sass lists font-awesome-sass as a runtime dependency of Patternfly 

cc - https://github.com/theforeman/foreman/pull/2943

### CODE REVIEW

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec [hunk 1]
```diff
@@ -0,0 +1,67 @@
+ %{?scl:%scl_package rubygem-%{gem_name}}
```
Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: remove the leading space

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+Release: 1%{?dist}
+Summary: Font-Awesome SASS
+Group: Development/Languages
+License: MIT
+URL: https://github.com/FortAwesome/font-awesome-sass
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+Requires: %{?scl_prefix_ruby}ruby
+BuildRequires: %{?scl_prefix_ruby}rubygem(sass-rails)
```
Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: ror SCL, not ruby?

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+BuildRequires: %{?scl_prefix_ruby}rubygem(compass)
```
Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: is this available?

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+BuildArch: noarch
+
+%description
+Font-Awesome SASS gem to be used in Ruby projects.
+
+%package doc
+Summary: Documentation for %{pkg_name}
+Group: Documentation
+Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
```
Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: no such package to obsolete

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/.gitignore
+%license %{gem_instdir}/LICENSE.txt
```
Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: %license will fail on EL6, it's a new attrs macro

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+%files doc
+%doc %{gem_docdir}
+%{gem_instdir}/Gemfile
+%doc %{gem_instdir}/README.md
+%{gem_instdir}/Rakefile
+%{gem_instdir}/font-awesome-sass.gemspec
+
+%changelog
+* Thu Dec 31 2015 Daniel Lobato - 4.3.2.1-1
```
Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: missing e-mail address

**Comment by domcleal** [comment]
General comments:

1) missing updates to comps/ and rel-eng/ to build the package, which is the first reason why tests will be failing.  Check steps at https://github.com/theforeman/foreman-packaging/tree/rpm/develop#howto-create-a-new-core-package-or-dependency.
2) missing require on rubygem(sass)
3) missing build requires on rubygem related tools, ruby(release) etc?
4) the build requires (e.g. sass, compass) don't seem to be used - you should either add a %check section to run tests or remove them

**Comment by domcleal** [comment]
> missing updates to comps/ and rel-eng/ to build the package

Oh, please rebase on rpm/develop before adding these as I renamed files from fedora19 to 21 this morning.

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec (hunk 3 above)
Comment (dLobatog) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: It doesn't even require it, my fault

File: comps/comps-foreman-fedora21.xml [hunk 7]
```diff
@@ -133,6 +135,8 @@
       <packagereq type="default">rubygem-jquery-turbolinks-doc</packagereq>
       <packagereq type="default">rubygem-jquery-ui-rails-doc</packagereq>
       <packagereq type="default">rubygem-jwt-doc</packagereq>
+      <packagereq type="default">rubygem-kafo-doc</packagereq>
```
Comment (dLobatog) [inline comps/comps-foreman-fedora21.xml]: comps_doc.sh added these automatically so I guess they were missing

**Comment by dLobatog** [comment]
Updated with the proper requires & comps

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+Release: 1%{?dist}
+Summary: Font-Awesome SASS
+Group: Development/Languages
+License: MIT
+URL: https://github.com/FortAwesome/font-awesome-sass
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+Requires: %{?scl_prefix_ruby}ruby
+Requires: %{?scl_prefix_ror}rubygem(sass) >= 3.2
```
Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: also `< 4.0` as it's `~> 3.2`

File: comps/comps-foreman-fedora21.xml (hunk 7 above)
Comment (domcleal) [inline comps/comps-foreman-fedora21.xml]: Yep, fixed in rpm/develop too now, thanks.

**Comment by domcleal** [comment]
Untested, but looks OK.  Will see it tested along with the other changes.

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec (hunk 8 above)
Comment (dLobatog) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: https://rubygems.org/gems/font-awesome-sass & https://github.com/FortAwesome/font-awesome-sass/blob/master/font-awesome-sass.gemspec say it's `>= 3.2`, where did you see the `~>`?

Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: On 4.3.2.1 specifically: https://rubygems.org/gems/font-awesome-sass/versions/4.3.2.1

**Comment by dLobatog** [comment]
@domcleal Thanks for the heads up on the rubygem-sass version, updated.

File: rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+Requires: %{?scl_prefix_ror}rubygem(sass) < 3.2
```
Comment (domcleal) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: `4.0`

Comment (dLobatog) [inline rubygem-font-awesome-sass/rubygem-font-awesome-sass.spec]: Updated, dunno what was I thinking

**Comment by domcleal** [comment]
Thanks @dLobatog, merged and building.

**Pull request merged**